### PR TITLE
added User-Agent header to default request headers

### PR DESCRIPTION
**Pull request by ayang99**
Storage service logs User-Agent received into audit log

### CODE REVIEW

File: dcc-storage-client/pom.xml [hunk 1]
```diff
@@ -160,9 +160,12 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF S
         <artifactId>maven-jar-plugin</artifactId>
         <configuration>
           <archive>
-            <manifestEntries>
-              <Implementation-Version>${project.version}</Implementation-Version>
-            </manifestEntries>
+
+            <manifest>
+              <mainClass>org.icgc.dcc.storage.client.ClientMain</mainClass>
```
Comment (btiernay) [inline dcc-storage-client/pom.xml]: This is not the main class and will break things. I suggest reverting the changes to this file.

File: dcc-storage-server/pom.xml [hunk 2]
```diff
@@ -156,6 +156,21 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF S
...
       </plugin>
+      
+      <!-- Jar -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+          <archive>
+            <manifest>
+              <mainClass>org.icgc.dcc.storage.server.ServerMain</mainClass>
```
Comment (btiernay) [inline dcc-storage-server/pom.xml]: Same issue here. This is not how Spring Boot works.

Comment (btiernay) [inline dcc-storage-server/pom.xml]: Actually, I may be wrong here. The main point is that the entry point is always `org.springframework.boot.loader.JarLauncher`

Comment (ayang99) [inline dcc-storage-server/pom.xml]: The spring-boot-maven-plugin manages the "main-class" and "start-class" entries anyways. So specifying those values in the maven-jar-plugin configuration is ignored. Manifest files always get the appropriate entries:
Main-Class: org.springframework.boot.loader.JarLauncher
Start-Class: org.icgc.dcc.storage.client.ClientMain

Comment (btiernay) [inline dcc-storage-server/pom.xml]: So why add `<mainClass>`? I think it is misleading / confusing to developers.

Comment (ayang99) [inline dcc-storage-server/pom.xml]: it's already been removed - when I was trying to get it to work, I thought I needed it.

**Pull request merged**